salt/states/ssh_auth.py

```diff
@@ -580,15 +580,31 @@ def manage(
         # gather list potential ssh keys for removal comparison
         # options, enc, and comments could be in the mix
         all_potential_keys.extend(ssh_key.split(" "))
-    existing_keys = __salt__["ssh.auth_keys"](user=user).keys()
+
+    existing_keys = __salt__["ssh.auth_keys"](
+        user=user, 
+        config=config, 
+        fingerprint_hash_type=fingerprint_hash_type
+    ).keys()
+
     remove_keys = set(existing_keys).difference(all_potential_keys)
     for remove_key in remove_keys:
         if __opts__["test"]:
             remove_comment = "{} Key set for removal".format(remove_key)
             ret["comment"] = remove_comment
             ret["result"] = None
         else:
-            remove_comment = absent(remove_key, user)["comment"]
+            remove_comment = absent(
+                remove_key, 
+                user,
+                enc,
```

Comment:
Could you pass these in as kwargs, for example `enc=enc`. Just future proofing in case any kwargs's position moves.

Reply:
Oop. Sorry for never getting back to this, notifications get spammed a lot by private repos. Glad to see this got picked and merged though!